src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -110,68 +110,193 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterOperationBlockStartAction(osContext =>
+            analysisContext.RegisterCompilationStartAction(compilationContext =>
             {
-                var method = osContext.OwningSymbol as IMethodSymbol;
-                if (method == null)
-                {
-                    return;
-                }
+                INamedTypeSymbol expectedExceptionType = WellKnownTypes.ExpectedException(compilationContext.Compilation);
```

Comment:
>expectedExceptionType [](start = 33, length = 21)

Consider even caching well known types for `NUnit.Framework.Assert` and `Xunit.Assert`, so you can bail out upfront from `ShouldSkipAnalyzing` if none of these types are defined.